src/pages/Browse/index.spec.tsx

```diff
@@ -358,4 +379,119 @@ describe(__filename, () => {
 
     expect(dispatchSpy).not.toHaveBeenCalled();
   });
+
+  it('does not dispatch anything on mount if a version is already loaded', () => {
+    const version = fakeVersion;
+
+    const store = configureStore();
+    _loadVersionAndFile({ store, version });
+    const dispatch = spyOn(store, 'dispatch');
+
+    render({ store, versionId: String(version.id) });
+
+    expect(dispatch).not.toHaveBeenCalled();
+  });
+
+  it('dispatches updateSelectedPath on mount if the query string contains a `path` that is different than `version.selectedPath`', () => {
+    const path = 'a/different/file.js';
+    const version = fakeVersion;
+    const history = createFakeHistory({
+      location: createFakeLocation({
+        search: queryString.stringify({ path }),
+      }),
+    });
+    const store = configureStore();
+    _loadVersionAndFile({ store, version });
+    const dispatch = spyOn(store, 'dispatch');
+
+    render({ store, versionId: String(version.id), history });
+
+    expect(dispatch).toHaveBeenCalledWith(
+      versionsActions.updateSelectedPath({
+        versionId: version.id,
+        selectedPath: path,
+      }),
+    );
+    expect(dispatch).toHaveBeenCalledTimes(1);
+  });
+
+  it('does not dispatch updateSelectedPath on mount when the query string does not contain a `path`', () => {
```

Comment:
Should there also be tests that assert that it does not dispatch `updateSelectedPath` if the path in the query string is the same as `version.selectedPath`, both for `mount` and `update`?